lower/llpcSpirvLowerResourceCollect.cpp

```diff
@@ -52,19 +52,102 @@ char SpirvLowerResourceCollect::ID = 0;
 
 // =====================================================================================================================
 // Pass creator, creates the pass of SPIR-V lowering opertions for resource collecting
-ModulePass* CreateSpirvLowerResourceCollect()
+ModulePass* CreateSpirvLowerResourceCollect(
+    bool collectDetailUsage) // Whether to collect detailed usages of resource node datas and FS output infos
 {
-    return new SpirvLowerResourceCollect();
+    return new SpirvLowerResourceCollect(collectDetailUsage);
 }
 
 // =====================================================================================================================
-SpirvLowerResourceCollect::SpirvLowerResourceCollect()
+SpirvLowerResourceCollect::SpirvLowerResourceCollect(
+    bool collectDetailUsage) // Whether to collect detailed usages of resource node datas and FS output infos
     :
-    SpirvLower(ID)
+    SpirvLower(ID),
+    m_collectDetailUsage(collectDetailUsage),
+    m_detailUsageValid(false)
 {
     initializeSpirvLowerResourceCollectPass(*PassRegistry::getPassRegistry());
 }
 
+// =====================================================================================================================
+// Collect resource node data
+void SpirvLowerResourceCollect::CollectResourceNodeData(
+    const GlobalVariable* pGlobal,       // Llvm global variable
```

Comment:
// Llvm global variable
->
// [in] Global variable to collect resource node data